callback-heaven.lisp

```diff
@@ -42,11 +49,11 @@
   (setf (gethash function-name (api-group-function-table group))
         new-value))
 
-(defmacro define-api-group (name)
+(defmacro define-api-group (name &optional (documentation nil docp))
```

Comment:
forgot to mention, the `&optional` here makes it hard to extend with additional keywords later if we wanted to. On the plus side, `&optional` is reminiscent of `defvar` / `defparameter`.

Reply:
Since it's a macro, I think we can keep the _syntax_ of the optional doc string. Doesn't mean we can't later `&body` it and `parse-body` later.